File: android/word/src/main/java/com/dreampany/word/data/source/remote/RemoteWordDataSource.java
/*
package com.dreampany.word.data.source.remote;

import android.graphics.Bitmap;

import com.annimon.stream.Stream;
import com.dreampany.frame.util.DataUtil;
import com.dreampany.network.manager.NetworkManager;
import com.dreampany.word.api.wordnik.WordnikManager;
import com.dreampany.word.api.wordnik.WordnikWord;
import com.dreampany.word.data.misc.WordMapper;
import com.dreampany.word.data.model.Word;
import com.dreampany.word.data.source.api.WordDataSource;
import com.dreampany.word.misc.Constants;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

import javax.inject.Singleton;

import io.reactivex.Maybe;
import timber.log.Timber;

*/
/**
 * Created by Hawladar Roman on 2/9/18.
 * Dreampany Ltd
 * dreampanymail@gmail.com
 *//*

@Singleton
public class RemoteWordDataSource implements WordDataSource {

    private final NetworkManager network;
    private final WordMapper mapper;
    private final WordnikManager wordnik;

    public RemoteWordDataSource(NetworkManager network,
                                WordMapper mapper,
                                WordnikManager wordnik) {
        this.network = network;
        this.mapper = mapper;
        this.wordnik = wordnik;
    }

    @Override
    public Word getTodayItem() {
        return null;
    }

    @Override
    public Maybe<Word> getTodayItemRx() {
        return null;
    }

    @Override
    public boolean isEmpty() {
        return false;
    }

    @Override
    public Maybe<Boolean> isEmptyRx() {
        return Maybe.fromCallable(this::isEmpty);
    }

    @Override
    public int getCount() {
        return 0;
    }

    @Override
    public Maybe<Integer> getCountRx() {
        return null;
    }

    @Override
    public boolean isExists(Word word) {
        return false;
    }

    @Override
    public Maybe<Boolean> isExistsRx(Word word) {
        return null;
    }

    @Override
    public long putItem(Word word) {
        return 0;
    }

    @Override
    public Maybe<Long> putItemRx(Word word) {
        return null;
    }

    @Override
    public int delete(Word word) {
        return 0;
    }

    @Override
    public Maybe<Integer> deleteRx(Word word) {
        return null;
    }

    @Override
    public Word getItem(String id) {
        return null;
    }

    @Override
    public Maybe<Word> getItemRx(String id) {
        return null;
    }

    @Override
    public Word getItem(String word, boolean full) {
        WordnikWord item = wordnik.getWord(word, Constants.Limit.WORD_RESOLVE);
        Timber.v("Wordnik Result %s", item.getWord());
        return mapper.toItem(word, item, true);
    }

    @Override
    public Maybe<Word> getItemRx(String word, boolean full) {
        return Maybe.fromCallable(() -> getItem(word, full));
    }

    @Override
    public List<Word> getItems(List<String> ids) {
        return null;
    }

    @Override
    public Maybe<List<Word>> getItemsRx(List<String> ids) {
        return null;
    }

    @Override
    public List<Word> getSearchItems(String query, int limit) {
        List<WordnikWord> items = wordnik.search(query, limit);
        List<Word> result = new ArrayList<>();
        if (!DataUtil.isEmpty(items)) {
            Stream.of(items).forEach(word -> result.add(mapper.toItem(word, false)));
        }
        return result;
    }

*/
/*    @Override
    public Maybe<List<Word>> getSearchItemsRx(String query) {
        return null;
    }

    @Override
    public Maybe<List<Word>> getSearchItemsRx(String query, int limit) {
        return Maybe.fromCallable(() -> {
            List<WordnikWord> items = wordnik.search(query, limit);
            List<Word> result = new ArrayList<>();
            if (!DataUtil.isEmpty(items)) {
                Stream.of(items).forEach(word -> result.add(mapper.toItem(word, false)));
            }
            return result;
        });
    }*//*


    @Override
    public List<Word> getCommonItems() {
        return null;
    }

    @Override
    public List<Word> getAlphaItems() {
        return null;
    }

    @Override
    public Maybe<List<Word>> getItemsRx(Bitmap bitmap) {
        return null;
    }

    @Override
    public List<String> getRawWords() {
        return null;
    }

    @Override
    public Maybe<List<String>> getRawWordsRx() {
        return null;
    }


    @NotNull
    @Override
    public List<Long> delete(@NotNull List<? extends Word> words) {
        return null;
    }

    @NotNull
    @Override
    public Maybe<List<Long>> deleteRx(@NotNull List<? extends Word> words) {
        return null;
    }

    @NotNull
    @Override
    public List<Long> putItems(@NotNull List<? extends Word> words) {
        return null;
    }

    @NotNull
    @Override
    public Maybe<List<Long>> putItemsRx(@NotNull List<? extends Word> words) {
        return null;
    }

    @NotNull
    @Override
    public List<Word> getItems() {
        return null;
    }

    @NotNull
    @Override
    public Maybe<List<Word>> getItemsRx() {
        return null;
    }

    @NotNull
    @Override
    public List<Word> getItems(int limit) {
        return null;
    }

    @NotNull
    @Override
    public Maybe<List<Word>> getItemsRx(int limit) {
        return null;
    }
}
*/

File: android/word/src/main/kotlin/com/dreampany/word/data/source/remote/RemoteWordDataSource.kt
package com.dreampany.word.data.source.remote

import android.graphics.Bitmap
import com.annimon.stream.Stream
import com.dreampany.framework.util.DataUtil
import com.dreampany.network.manager.NetworkManager
import com.dreampany.word.api.wordnik.WordnikManager
import com.dreampany.word.data.misc.WordMapper
import com.dreampany.word.data.model.Word
import com.dreampany.word.data.source.api.WordDataSource
import com.dreampany.word.misc.Constants
import io.reactivex.Maybe
import timber.log.Timber
import java.util.*
import javax.inject.Singleton

/**
 * Created by roman on 2019-07-13
 * Copyright (c) 2019 bjit. All rights reserved.
 * hawladar.roman@bjitgroup.com
 * Last modified $file.lastModified
 */

@Singleton
class RemoteWordDataSource(
    val network: NetworkManager,
    val mapper: WordMapper,
    val wordnik: WordnikManager
) : WordDataSource {
    override fun isExists(word: String): Boolean {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getTodayItem(): Word {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getTodayItemRx(): Maybe<Word> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getItem(word: String, full: Boolean): Word? {
        val item = wordnik.getWord(word, Constants.Limit.WORD_RESOLVE)
        Timber.v("Wordnik Result %s", item!!.word)
        return mapper.toItem(word, item, true)
    }

    override fun deleteRx(t: Word): Maybe<Int> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getItem(id: String): Word? {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getItemRx(word: String, full: Boolean): Maybe<Word> {
        return Maybe.fromCallable { getItem(word, full) }
    }

    override fun getItemRx(id: String): Maybe<Word> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getItems(ids: List<String>): List<Word> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getItems(): List<Word> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getItems(limit: Long): List<Word> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getItemsRx(ids: List<String>): Maybe<List<Word>> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getItemsRx(bitmap: Bitmap): Maybe<List<Word>> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getItemsRx(): Maybe<List<Word>> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getItemsRx(limit: Int): Maybe<List<Word>> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getSearchItems(query: String, limit: Int): List<Word> {
        val items = wordnik.search(query, limit)
        val result = ArrayList<Word>()
        if (!DataUtil.isEmpty(items)) {
            Stream.of(items!!).forEach { word -> result.add(mapper.toItem(word, false)) }
        }
        return result
    }

    override fun getCommonItems(): List<Word> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getAlphaItems(): List<Word> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getRawWords(): List<String> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getRawWordsRx(): Maybe<List<String>> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun isEmpty(): Boolean {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun isEmptyRx(): Maybe<Boolean> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getCount(): Int {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getCountRx(): Maybe<Int> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun isExists(t: Word): Boolean {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun isExistsRx(t: Word): Maybe<Boolean> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun putItem(t: Word): Long {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun putItemRx(t: Word): Maybe<Long> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun putItems(ts: List<Word>): List<Long> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun putItemsRx(ts: List<Word>): Maybe<List<Long>> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun delete(t: Word): Int {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun delete(ts: List<Word>): List<Long> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun deleteRx(ts: List<Word>): Maybe<List<Long>> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }
}